test/dlc_tests/benchmark/sagemaker/tensorflow/training/test_trcomp_performance.py

```diff
@@ -0,0 +1,175 @@
+    if 'gpu' not in tensorflow_training:
+        pytest.skip('Training Compiler is only available for GPUs')
+
+
+
+def pytest_generate_tests(metafunc):
+    if "instance_type" in metafunc.fixturenames:
+        metafunc.parametrize("instance_type, instance_count", [
+                                                pytest.param('ml.p3.2xlarge', 1, marks=[pytest.mark.p3, pytest.mark.single_gpu]),
+                                                # pytest.param('ml.p3.16xlarge', 1, marks=[pytest.mark.p3, pytest.mark.single_node_multi_gpu]),
+                                                # pytest.param('ml.p3dn.24xlarge', 2, marks=[pytest.mark.p3, pytest.mark.multi_node_multi_gpu]),
+                                                ])
+
+
+
+@pytest.mark.flaky(reruns=1)
+@pytest.mark.usefixtures("sagemaker_only")
+class TestImageClassification:
+
+
+    @pytest.mark.model("resnet101")
+    def test_resnet101_at_fp16(self, instance_type, num_gpus, total_n_gpus, instance_count, distribution_strategy, caching, tensorflow_training, sagemaker_session, capsys, framework_version):
+        epochs = int(100*total_n_gpus)
+        batches = np.array([224])*total_n_gpus
+        for batch in np.array(batches, dtype=int):
+            train_steps = int(10240*epochs/batch)
+            steps_per_loop = train_steps//10
+            overrides=\
+            f"runtime.enable_xla=True,"\
+            f"runtime.num_gpus={num_gpus},"\
+            f"runtime.distribution_strategy={distribution_strategy},"\
+            f"runtime.mixed_precision_dtype=float16,"\
+            f"task.train_data.global_batch_size={batch},"\
+            f"task.train_data.input_path=/opt/ml/input/data/training/validation*,"\
+            f"task.train_data.cache={caching},"\
+            f"trainer.train_steps={train_steps},"\
+            f"trainer.steps_per_loop={steps_per_loop},"\
+            f"trainer.summary_interval={steps_per_loop},"\
+            f"trainer.checkpoint_interval={train_steps},"\
+            f"task.model.backbone.type=resnet,"\
+            f"task.model.backbone.resnet.model_id=101"
+            estimator = TensorFlow(
+                                sagemaker_session=sagemaker_session,
+                                git_config={
+                                    'repo': 'https://github.com/tensorflow/models.git',
+                                    'branch': 'v2.9.2',
+                                },
+                                source_dir='.',
+                                entry_point='official/vision/train.py',
+                                model_dir=False,
+                                instance_type=instance_type,
+                                instance_count=instance_count,
+                                image_uri=tensorflow_training,
+                                hyperparameters={
+                                    TrainingCompilerConfig.HP_ENABLE_COMPILER : True,
```

Comment:
Why do we have to enable it like this? What does this mean?

Reply:
Since we do not have the SDK changes yet, this is a workaround to test this release. Once the SDK changes are released, we can do ```compiler_config=TrainingCompilerConfig()```